### PR TITLE
More standard library documentation updates.

### PR DESCRIPTION
**Pull request by yole**

### CODE REVIEW

File: libraries/stdlib/src/kotlin/collections/JUtil.kt [hunk 1]
```diff
@@ -34,33 +34,38 @@ private object EmptySet : Set<Any> {
...
-/** Returns a new read-only ordered set of given elements */
+/** Returns a new read-only ordered set with the given elements. */
 public fun setOf<T>(vararg values: T): Set<T> = if (values.size() == 0) emptySet() else values.toCollection(LinkedHashSet<T>())
 
-/** Returns an empty read-only set */
+/** Returns an empty read-only set. */
 public fun setOf<T>(): Set<T> = emptySet()
 
-/** Returns a new LinkedList with a variable number of initial elements */
+/** Returns a new LinkedList with the given elements. */
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/collections/JUtil.kt]: Linkify LinkedList

File: libraries/stdlib/src/kotlin/collections/JUtil.kt [hunk 2]
```diff
@@ -34,33 +34,38 @@ private object EmptySet : Set<Any> {
...
-/** Returns a new HashSet with a variable number of initial elements */
+/** Returns a new HashSet with the given elements. */
 public fun hashSetOf<T>(vararg values: T): HashSet<T> = values.toCollection(HashSet(values.size()))
 
-/** Returns a new LinkedHashSet with a variable number of initial elements */
+/** Returns a new LinkedHashSet with the given elements. */
 public fun linkedSetOf<T>(vararg values: T): LinkedHashSet<T> = values.toCollection(LinkedHashSet(values.size()))
 
+/**
+ * Returns the valid indices for this collection.
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/collections/JUtil.kt]: Returns an IntRange specifying the valid...

File: libraries/stdlib/src/kotlin/concurrent/Locks.kt [hunk 3]
```diff
@@ -5,8 +5,8 @@ import java.util.concurrent.locks.ReentrantReadWriteLock
 import java.util.concurrent.CountDownLatch
 
 /**
- * Executes given calculation under lock
- * Returns result of the calculation
+ * Executes the given calculation under this lock.
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/concurrent/Locks.kt]: calculation -> action

File: libraries/stdlib/src/kotlin/io/IOStreams.kt [hunk 4]
```diff
@@ -13,39 +13,45 @@ public fun InputStream.iterator(): ByteIterator =
...
+ * @param bufferSize the buffer size to use.
+ */
 public fun InputStream.buffered(bufferSize: Int = defaultBufferSize): InputStream
         = if (this is BufferedInputStream)
     this
 else
     BufferedInputStream(this, bufferSize)
 
-/** Creates a reader on an input stream using UTF-8 or specified charset. */
+/** Creates a reader on this input stream using UTF-8 or the specified [charset]. */
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/io/IOStreams.kt]: Need to put it in conventions, wording around documenting parameter with default value.

File: libraries/stdlib/src/kotlin/io/ReadWrite.kt [hunk 5]
```diff
@@ -139,29 +140,34 @@ public fun File.readLines(charset: Charset = Charsets.UTF_8): List<String> {
     return result
 }
 
-/** Creates a buffered reader, or returns self if Reader is already buffered */
+/** Creates a buffered reader wrapping this Reader, or returns self if Reader is already buffered */
 public fun Reader.buffered(bufferSize: Int = defaultBufferSize): BufferedReader
         = if (this is BufferedReader) this else BufferedReader(this, bufferSize)
 
-/** Creates a buffered writer, or returns self if Writer is already buffered */
+/** Creates a buffered writer wrapping this Writer, or returns self if Writer is already buffered */
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/io/ReadWrite.kt]: self -> this Writer

File: libraries/stdlib/src/kotlin/io/ReadWrite.kt [hunk 6]
```diff
@@ -139,29 +140,34 @@ public fun File.readLines(charset: Charset = Charsets.UTF_8): List<String> {
     return result
 }
 
-/** Creates a buffered reader, or returns self if Reader is already buffered */
+/** Creates a buffered reader wrapping this Reader, or returns self if Reader is already buffered */
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/io/ReadWrite.kt]: self -> this Reader

File: libraries/stdlib/src/kotlin/jvm/JvmFlagAnnotations.kt [hunk 7]
```diff
@@ -19,18 +19,39 @@ package kotlin.jvm
 import java.lang.annotation.Retention
 import java.lang.annotation.RetentionPolicy
 
+/**
+ * Marks the JVM backing field of the annotated property as `volatile`, meaning that it can
+ * be modified asynchronously by concurrently running threads.
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/jvm/JvmFlagAnnotations.kt]: "can be modified async" is strange wording. Even if field is not volatile, you still can modify it async by concurrent threads, but effect may be different. Need better description.

File: libraries/stdlib/src/kotlin/text/StringBuilderJVM.kt [hunk 8]
```diff
@@ -14,48 +14,47 @@ public fun Appendable.appendln(value: CharSequence?): Appendable = append(value)
 /** Appends value to the given Appendable and line separator after it. */
 public fun Appendable.appendln(value: Char): Appendable = append(value).append(LINE_SEPARATOR)
 
-/** Appends line separator to StringBuilder. */
-
+/** Appends a line separator to this StringBuilder. */
 public fun StringBuilder.appendln(): StringBuilder = append(LINE_SEPARATOR)
 
-/** Appends value to the given StringBuilder and line separator after it. */
+/** Appends value to this StringBuilder and line separator after it. */
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/text/StringBuilderJVM.kt]: backtickify value

File: libraries/stdlib/src/kotlin/collections/JUtil.kt (hunk 1 above)
Comment (yole) [inline libraries/stdlib/src/kotlin/collections/JUtil.kt]: done

File: libraries/stdlib/src/kotlin/collections/JUtil.kt (hunk 2 above)
Comment (yole) [inline libraries/stdlib/src/kotlin/collections/JUtil.kt]: done

File: libraries/stdlib/src/kotlin/concurrent/Locks.kt (hunk 3 above)
Comment (yole) [inline libraries/stdlib/src/kotlin/concurrent/Locks.kt]: done

File: libraries/stdlib/src/kotlin/io/IOStreams.kt (hunk 4 above)
Comment (yole) [inline libraries/stdlib/src/kotlin/io/IOStreams.kt]: Any specific wording you'd like to suggest?

File: libraries/stdlib/src/kotlin/io/ReadWrite.kt (hunk 5 above)
Comment (yole) [inline libraries/stdlib/src/kotlin/io/ReadWrite.kt]: done

File: libraries/stdlib/src/kotlin/io/ReadWrite.kt (hunk 6 above)
Comment (yole) [inline libraries/stdlib/src/kotlin/io/ReadWrite.kt]: done

File: libraries/stdlib/src/kotlin/jvm/JvmFlagAnnotations.kt (hunk 7 above)
Comment (yole) [inline libraries/stdlib/src/kotlin/jvm/JvmFlagAnnotations.kt]: Made an attempt. This is in any case not a suitable place for explaining the Java memory model in detail.

File: libraries/stdlib/src/kotlin/text/StringBuilderJVM.kt (hunk 8 above)
Comment (yole) [inline libraries/stdlib/src/kotlin/text/StringBuilderJVM.kt]: done

**Pull request merged**